Fixed the pagerando specific tests.

Saldivarcher opened the pull request:
This is a fix for the `pagerando` tests that seemed to be failing due to syntax change, and other various reasons.

My main concern was for the `varargs` test, `opt` seemed to be putting the functions in differing places like so:
```
define void @real_va_type(...) #0 {
  %1 = alloca %struct.va_list
  %2 = bitcast %struct.va_list* %1 to i8*
  call void @llvm.va_start(i8* %2)
  call void @"real_va_type$$origva"(%struct.va_list* %1)
  %3 = bitcast %struct.va_list* %1 to i8*
  call void @llvm.va_end(i8* %3)
  ret void
}

; Function Attrs: noinline optsize
define void @copy(...) #0 {
  %1 = alloca i8
  call void @llvm.va_start(i8* %1)
  call void @"copy$$origva"(i8* %1)
  call void @llvm.va_end(i8* %1)
  ret void
}

; Function Attrs: noinline optsize
define void @multiple_starts(...) #0 {
  %1 = alloca i8
  call void @llvm.va_start(i8* %1)
  call void @"multiple_starts$$origva"(i8* %1)
  call void @llvm.va_end(i8* %1)
  ret void
}

; Function Attrs: noinline optsize
define i32 @varags(...) #0 {
  %1 = alloca i8
  call void @llvm.va_start(i8* %1)
  %2 = call i32 @"varags$$origva"(i8* %1)
  call void @llvm.va_end(i8* %1)
  ret i32 %2
}
```
So, is using `CHECK-DAG`, the correct way of handling that?

Saldivarcher committed:
Fixed the pagerando specific tests.
Tests either had very little changes to significant ones due to:
syntax change, function attributes, and other minor issues.

Change-Id: Ibc29e3c9b87bc5cd6e43bf6049a3486b464250a4

rinon commented:
I haven't gone through closely, will do that later. However, I can tell you right now that CHECK-DAG isn't what you want for the varargs. All CHECK-DAG lines can be re-ordered, until hitting a non-DAG check. So... in this case, using CHECK-DAG on all the lines means they might be seen in any order, which seems to loose of a constraint to me.

Saldivarcher committed:
Made the varargs test a bit stricter

Change-Id: I38deb5692f0e33a45895c3ec1893b0f8de4fb3c5

rinon commented on test/CodeGen/AArch64/pagerando-isel.ll:
```diff
@@ -5,48 +5,48 @@
+; CHECK-DAG: [[POT:%[0-9]+]]:gpr64 = COPY $x20
+; CHECK-DAG: [[GOT:%[0-9]+]]:gpr64common = LOADpot [[POT]]:gpr64, 0
+
+; CHECK-DAG: %{{[0-9]+}}:gpr64 = MOVZXi target-flags(aarch64-g0, aarch64-nc, <unknown bitmask target flag>) @legacy, 0
```
We should probably fix the unknown bitmask target flags. Should be a separate commit though.

rinon commented on test/CodeGen/ARM/pagerando-isel.ll:
```diff
@@ -5,47 +5,51 @@
-; CHECK: [[BINNED_BINOFF:%vreg[0-9]+]]<def> = LDRi12 <cp#3>, 0, pred:14, pred:%noreg
-; CHECK: [[BINNED:%vreg[0-9]+]]<def> = ADDrr [[BINNED_BIN]]<kill>, [[BINNED_BINOFF]]<kill>, pred:14, pred:%noreg, opt:%noreg
-; CHECK: BLX [[BINNED]]<kill>
+  
```
Trailing whitespace

rinon replied:
Remove this whitespace and the changes LGTM.

rinon commented on test/CodeGen/AArch64/pagerando-isel.ll:
```diff
@@ -5,48 +5,48 @@
 define void @legacy() { ret void }
-define void @wrapper() pagerando_wrapper { ret void }
-define hidden void @binned() pagerando_binned { ret void }
+define void @wrapper() pagerando { ret void }
```
Wrapper functions don't get the pagerando attribute applied (see the PagerandoWrappers tests). This applies to everything called a "wrapper." The backend specific tests are after binning, so we are simulating a wrapper (in .text) and a wrapped, binned function.

Saldivarcher replied:
Okay, I get it now. Thank you for the explanation :smile:

rinon commented on test/Transforms/PagerandoWrappers/varargs.ll:
```diff
@@ -1,56 +1,68 @@
   ret void
 }
+; CHECK-LABEL: define void @real_va_type(...)
+; CHECK:    %1 = alloca %struct.va_list
```
Why did you change `CHECK-NEXT` into `CHECK` directives? If there's a reason that's ok, but I just would like to know why.

Saldivarcher replied:
Oops, you're right it should be `CHECK-NEXT`.

rinon commented on test/CodeGen/ARM/pagerando-optimizer.ll:
```diff
@@ -1,15 +1,17 @@
+; CHECK-LABEL: .type wrapper
 ; CHECK-LABEL: wrapper:
-define void @wrapper() pagerando_wrapper { ret void }
+define void @wrapper() pagerando { ret void }
```
Again, this shouldn't have the pagerando attribute (see above).

rinon commented on test/CodeGen/ARM/pagerando-optimizer.ll:
```diff
@@ -1,15 +1,17 @@
 
-; CHECK-LABEL: .text
+; CHECK-LABEL: .section .text.bin_1
+; CHECK-LABEL: .type wrapper
```
I don't think we need these `.type` lines. I think they're for type-based CFI?

rinon commented on test/CodeGen/AArch64/pagerando-optimizer.ll:
```diff
@@ -1,19 +1,18 @@
+; CHECK-LABEL: section .text.bin_1
 ; CHECK-LABEL: wrapper:
-define void @wrapper() pagerando_wrapper { ret void }
+define void @wrapper() pagerando { ret void }
```
No pagerando attr (see above)

rinon reviewed (commented):
Almost good to go, just a few things to clean up.

Saldivarcher committed:
Fixed various issues brought up in a review.

Changed `CHECK` to `CHECK-NEXT` in the varargs test.
Removed some whitespace from the `pagerando` tests.
Removed the `pagerando` function attribute from the wrappers.

Change-Id: I5c581f9f77af623a7a27e8b59c7b2846c4cd77f1

rinon commented on test/Transforms/PagerandoWrappers/varargs.ll:
```diff
@@ -1,18 +1,39 @@
 ; RUN: opt < %s -pagerando-wrappers -S | FileCheck %s
 
+; CHECK-LABEL: %struct.va_list = type { i8*, i8*, i8*, i32, i32 }
```
No need to add this. We want the test to be as generic and minimal as possible while still checking for the functionality we're testing.

Saldivarcher committed:
Removed a check that was not needed

Change-Id: I55deb5890a405bf2d68af64cbf2eda5b2fdd0f58

Saldivarcher committed:
Removed extra whitespace
